src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -427,112 +245,92 @@ func Register(
 	return handleRegistrationFlow(req, r, sessionID, cfg, accountDB, deviceDB)
 }
 
-// handleRegistrationFlow will direct and complete registration flow stages
-// that the client has requested.
-func handleRegistrationFlow(
-	req *http.Request,
-	r registerRequest,
-	sessionID string,
-	cfg *config.Dendrite,
-	accountDB *accounts.Database,
-	deviceDB *devices.Database,
-) util.JSONResponse {
-	// TODO: Shared secret registration (create new user scripts)
-	// TODO: Enable registration config flag
-	// TODO: Guest account upgrading
-
-	// TODO: Handle loading of previous session parameters from database.
-	// TODO: Handle mapping registrationRequest parameters into session parameters
-
-	// TODO: email / msisdn auth types.
-
-	if cfg.Matrix.RegistrationDisabled && r.Auth.Type != authtypes.LoginTypeSharedSecret {
-		return util.MessageResponse(http.StatusForbidden, "Registration has been disabled")
-	}
-
-	switch r.Auth.Type {
-	case authtypes.LoginTypeRecaptcha:
-		// Check given captcha response
-		resErr := validateRecaptcha(cfg, r.Auth.Response, req.RemoteAddr)
-		if resErr != nil {
-			return *resErr
-		}
-
-		// Add Recaptcha to the list of completed registration stages
-		sessions.AddCompletedStage(sessionID, authtypes.LoginTypeRecaptcha)
-
-	case authtypes.LoginTypeSharedSecret:
+func handleLoginTypes(req *http.Request, r registerRequest, cfg *config.Dendrite, sessionID string) *util.JSONResponse {
```

Comment:
this is an odd name for this function, given it only handles LoginTypeSharedSecret. Please rename it.